### PR TITLE
feat(www): preliminary support for uploading to the shared storage

### PR DESCRIPTION
**Pull request by chombourger**

### CODE REVIEW

File: mtda/templates/index.html [hunk 1]
```diff
@@ -235,11 +266,75 @@
...
+                uploadSelectedFile();
+              }
+            }
+            break;
           default:
             storage_status.innerHTML = "unknown_document"
             storage_status.title = "storage status unknown"
         }
       });
+      const CHUNK_SIZE = 512 * 1024;
```
Comment (fmoessbauer) [inline mtda/templates/index.html:282]: What is the unit? Bytes?

**Review by fmoessbauer — COMMENTED** [review]
In general this looks quite good (and I love to see that feature). I'm just wondering what happens if the upload is interrupted or gets stuck. Does this end-up in a non-recoverable situation as the storage remains open?

**Comment by chombourger** [comment]
> In general this looks quite good (and I love to see that feature). I'm just wondering what happens if the upload is interrupted or gets stuck. Does this end-up in a non-recoverable situation as the storage remains open?

Reworked a session handling code a bit and will now close the shared storage device if the owning session has vanished

File: mtda/templates/index.html (hunk 1 above)
Comment (chombourger) [reply mtda/templates/index.html:282]: yes bytes. I need to test this feature a little more and in particular with large images. May even need to throttle the upload a bit as the browser will probably read the image much faster than we can write it. I have entered new territory while trying to implement this feature

**Pull request merged**